shotover-proxy/src/transforms/redis/cache.rs

```diff
@@ -74,386 +296,493 @@ impl SimpleRedisCache {
+                    column_name, relation_elements[0].value
+                );
 
-            insert_or_update(table_cache_schema, query_values, pk)
+                key.push(b':');
+                key.extend(relation_elements[0].value.to_string().as_bytes());
+            }
         }
-        statement => Err(anyhow!("Cant build query from statement: {}", statement)),
     }
+    Ok(Bytes::from(key))
 }
 
-fn insert_or_update(
+/// build the redis key for the query.
+/// key is cassandra partition key (must be completely specified) prepended to
+/// the cassandra range key (may be partially specified)
+fn build_query_redis_hash_from_value_map(
     table_cache_schema: &TableCacheSchema,
-    query_values: HashMap<String, &Value>,
-    pk: BytesMut,
-) -> Result<RedisFrame> {
-    let mut commands_buffer: Vec<RedisFrame> = vec![
-        RedisFrame::BulkString("ZADD".into()),
-        RedisFrame::BulkString(pk.freeze()),
-    ];
-
-    let clustering = table_cache_schema
-        .range_key
-        .iter()
-        .map(|k| query_values.get(k.as_str()).unwrap())
-        .fold(BytesMut::new(), |mut acc, v| {
-            acc.extend(MessageValue::from(*v).into_str_bytes());
-            acc
-        });
-
-    let values = query_values
-        .iter()
-        .filter_map(|(p, v)| {
-            if table_cache_schema.partition_key.iter().all(|x| x != p)
-                && table_cache_schema.range_key.iter().all(|x| x != p)
-            {
-                Some(MessageValue::from(*v))
-            } else {
-                None
-            }
-        })
-        .collect_vec();
-
-    for v in values {
-        commands_buffer.push(RedisFrame::BulkString(Bytes::from_static(b"0")));
-        let mut value = clustering.clone();
-        if !value.is_empty() {
-            value.put_u8(b':');
-        }
-        value.extend(v.clone().into_str_bytes());
-        commands_buffer.push(RedisFrame::BulkString(value.freeze()));
+    query_values: &BTreeMap<String, Vec<RelationElement>>,
+    select: &Select,
+) -> Result<Bytes, CacheableState> {
+    let mut my_values = query_values.clone();
+    for c_name in &table_cache_schema.partition_key {
+        let column_name = c_name.to_lowercase();
+        my_values.remove(&column_name);
+    }
+    for c_name in &table_cache_schema.range_key {
+        let column_name = c_name.to_lowercase();
+        my_values.remove(&column_name);
     }
 
-    Ok(RedisFrame::Array(commands_buffer))
+    let mut str = if select.columns.is_empty() {
+        String::from("WHERE ")
+    } else {
+        let mut tmp = select
+            .columns
+            .iter()
+            .map(|select_element| match select_element {
+                SelectElement::Star => SelectElement::Star,
+                SelectElement::Column(named) => SelectElement::Column(Named {
+                    name: named.name.to_lowercase(),
+                    alias: named.alias.as_ref().map(|name| name.to_lowercase()),
```

Comment:
While researching why we do this I found this page: https://docs.datastax.com/en/cql-oss/3.x/cql/cql_reference/ucase-lcase_r.html
Looks like we'll need even more complicated logic :worried: 
We need to:
* strip out the `"`
* replace the `""` escape sequence with `"` 
* handle the `"` case as non case sensitive.

It looks like we'll need to do the same for table names and column names